### PR TITLE
automate cloudant api key management

### PR DESCRIPTION
**Pull request by dannyroberts**
Assuming your localsettings is set to the same db prefix as you're trying to create on cloudant (e.g. `commcarehq` or `staging_commcarehq`

Create all necessary databases

```
python scripts/cloudant/create_databases.py --databases `./manage.py list_couchdbs` --username dimagi-danny
```

Generate a new api key and add it to all necessary dbs

```
python scripts/cloudant/generate_api_key.py --databases `./manage.py list_couchdbs` --username dimagi-danny
```

Revoke a api key from all necessary dbs

```
python scripts/cloudant/revoke_api_key.py mynolongernecessaryapikey --databases `./manage.py list_couchdbs` --username dimagi-danny
```

And for completeness, if you already have an API (and don't want to create a new one), add it to all necessary dbs

```
python scripts/cloudant/grant_api_key.py myexistingapikey --databases `./manage.py list_couchdbs` --username dimagi-danny
```

### CODE REVIEW

**Comment by dannyroberts** [comment]
@proteusvacuum cc @calellowitz

**Comment by czue** [comment]
nice! :closed_lock_with_key:

File: hqscripts/management/commands/list_couchdbs.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+from django.core.management.base import BaseCommand
+from corehq.util.couchdb_management import couch_config
+
+
+class Command(BaseCommand):
+    """
+    Deletes sofabed cruft from couchlog.
```
Comment (proteusvacuum) [inline hqscripts/management/commands/list_couchdbs.py]: Leftover from Copy-Pasta?

File: scripts/cloudant/manage_cloudant.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        if ask:
+            ask_user('Generating new API key for {}'.format(self.username))
+        new_api_key_response = requests.post(
+            'https://{username}.cloudant.com/_api/v2/api_keys'.format(username=self.username),
+            auth=self).json()
+        assert new_api_key_response['ok'] is True
+        return Auth(new_api_key_response['key'], new_api_key_response['password'])
+
+
+class CloudantDatabase(namedtuple('CloudantInstance', 'instance db_name')):
```
Comment (proteusvacuum) [inline scripts/cloudant/manage_cloudant.py]: Cool! I haven't really seen this pattern in use before, but it is pretty neat.

File: settingshelper.py [hunk 3]
```diff
@@ -192,8 +193,8 @@ def fix_logger_obfuscation(fix_logger_obfuscation_, logging_config):
         for handler in logging_config["handlers"].values():
             if handler["class"].startswith("corehq."):
                 if fix_logger_obfuscation_ != 'quiet':
-                    print "{} logger is being changed to {}".format(
+                    print("{} logger is being changed to {}".format(
                         handler['class'],
                         'logging.StreamHandler'
-                    )
+                    ), file=sys.stderr)
```
Comment (proteusvacuum) [inline settingshelper.py]: What's the rationale behind this change?

**Comment by proteusvacuum** [comment]
:+1: Neat! I think you could probably add the info in your PR comment into a `README.md ` under `scripts/cloudant`, or somewhere similar, so that it is easy to reference.

**Comment by proteusvacuum** [comment]
No blocking comments, so merging.

**Comment by snopoke** [comment]
Do you see this as a tool for managing you're personal cloudant DB's or production ones?

**Comment by dannyroberts** [comment]
@snopoke production dbs. This is to make it really easy to switch to API keys and then rotate them on whatever frequency we decided.

File: settingshelper.py (hunk 3 above)
Comment (dannyroberts) [inline settingshelper.py]: It was printing stuff to stdout when I ran `./manage.py list_couchdbs` and that was preventing me from using the output of that command to specify the dbs for the other command.

File: hqscripts/management/commands/list_couchdbs.py (hunk 1 above)
Comment (dannyroberts) [inline hqscripts/management/commands/list_couchdbs.py]: Thanks, fixing

**Comment by dannyroberts** [comment]
@proteusvacuum good idea, I'll add the readme.
